// option optimize_for = LITE_RUNTIME;

message DashDotProto
{
  repeated double dd = 1;
}

message LineRuleProto
{
  required double width = 1;
  required int32 color = 2;
  optional DashDotProto dashdot = 3;
}

message AreaRuleProto
{
  required int32 color = 1;
  optional LineRuleProto border = 2;
}

message SymbolRuleProto
{
  required string name = 1;
}

message CaptionRuleProto
{
  required int32 height = 1;
  optional int32 color = 2;
  optional int32 stroke_color = 3;
}

message CircleRuleProto
{
  required double radius = 1;
  required int32 color = 2;
  optional LineRuleProto border = 3;
}

// PathTextRule is same as CaptionRule

// WayMarkerRule not used yet


// Describe containers.

message DrawElementProto
{
  required int32 scale = 1;
  repeated LineRuleProto lines = 2;
  optional AreaRuleProto area = 3;
  optional SymbolRuleProto symbol = 4;
  optional CaptionRuleProto caption = 5;
  optional CircleRuleProto circle = 6;
}

message ClassifElementProto
{
  required string name = 1;
  repeated DrawElementProto element = 2;
}

message ContainerProto
{
  repeated ClassifElementProto cont = 1;
}
